runtime/vm/JFRConstantPoolTypes.hpp

```diff
@@ -1246,6 +1270,39 @@ class VM_JFRConstantPoolTypes {
 		gcConfiguration->heapAddressBits = J9JAVAVM_REFERENCE_SIZE(vm) * 8;
 	}
 
+	static uintptr_t recordSystemProcessEvent(uintptr_t pid, const char *commandLine, void *userData)
+	{
+		VM_JFRConstantPoolTypes *constantPoolTypes = reinterpret_cast<VM_JFRConstantPoolTypes *>(userData);
+		PORT_ACCESS_FROM_JAVAVM(constantPoolTypes->_vm);
+		J9Pool *systemProcessTable = constantPoolTypes->getSystemProcessTable();
+		UDATA cmdLength = strlen(commandLine);
+		char *commandLineCopy = reinterpret_cast<char *>(j9mem_allocate_memory(cmdLength + 1, OMRMEM_CATEGORY_VM));
+		if (NULL == commandLineCopy) {
+			return ~(uintptr_t)0;
+		}
+		memcpy(commandLineCopy, commandLine, cmdLength + 1);
+		SystemProcessEntry *entry = reinterpret_cast<SystemProcessEntry *>(pool_newElement(systemProcessTable));
+		if (NULL == entry) {
+			j9mem_free_memory(commandLineCopy);
+			return ~(uintptr_t)0;
+		}
+		entry->ticks = j9time_nano_time();
+		entry->commandLine = commandLineCopy;
```

Comment:
The `memcpy()` can wait until we're sure we can create a new pool entry.
The entry fields of should assigned (together) in order.
```c
		SystemProcessEntry *entry = reinterpret_cast<SystemProcessEntry *>(pool_newElement(systemProcessTable));
		if (NULL == entry) {
			j9mem_free_memory(commandLineCopy);
			return ~(uintptr_t)0;
		}
		memcpy(commandLineCopy, commandLine, cmdLength + 1);
		entry->ticks = j9time_nano_time();
		entry->pid = pid;
		entry->commandLine = commandLineCopy;
```